frontend/javascripts/dashboard/folders/folder_tree.tsx

```diff
@@ -0,0 +1,306 @@
+import React, { useEffect, useState } from "react";
+import { DropTargetMonitor, useDrop } from "react-dnd";
+import SortableTree, {
+  ExtendedNodeData,
+  FullTree,
+  NodeData,
+  OnDragPreviousAndNextLocation,
+  OnMovePreviousAndNextLocation,
+} from "react-sortable-tree";
```

Comment:
I looked into this today and the initial experience with the antd module was quite pleasant. However, I ran into a roadblock: I couldn't get dragging a dataset from the table into the sidebar to work. Apparently, antds own drag-and-drop behavior (for moving folders) prevents the custom "drop target" from working. If I disable the `draggable` property, dragging datasets to folders works again, but obviously moving folders doesn't work then, anymore.

I found the following related issues:

https://github.com/react-component/tree/issues/376
https://github.com/react-component/tree/issues/419
https://github.com/react-component/tree/issues/443
https://github.com/react-component/tree/issues/470

So, it seems to be known, but not being really actively worked on. Maybe we could find a workaround without touching the library code, but I don't really know how this could be done, since I don't know how the drag/drop events are intercepted in the first place.

I pushed my changes to `folders-antd`, but unless you have a good idea, I would think that it's not really worth to investigate further right now :( It's a bummer, since the styling really looked far better.. Maybe I'll adapt the styling a bit manually, so that we can draw some benefits from antd in the end.

Reply:
Argh, that's too bad :( Thank you for trying! Indeed, the issues and the one PR look effectively abandoned. Also, the code in the PR looks like a larger change so I am not too optimistic that there's an easy solution. I'll have a quick look at the `folders-antd` branch, though :)

Reply:
Ok, after reading my comment from above I had an idea how to workaround this issue! I don't want to toot my own horn, but I think it's pretty great (and simple) :smile: See https://github.com/scalableminds/webknossos/commit/7cdae93cd934b4a217ab791ab5be124085255efd. The idea is that folder DnD and dataset DnD are effectively incompatible, but they don't need to be compatible **at the same time**. So, as soon as a dataset is being dragged, I disable the tree's DnD behavior (and luckily, this information is already available :)).

I don't know whether there will be other roadblocks, but I think the critical stuff is working then (folder + dataset DnD, context menu).

Reply:
Uuh nice, that would be a great workaround, fingers crossed :grin: